index.js

```diff
@@ -1 +1,21 @@
-// Do work!
\ No newline at end of file
+/* Do work! Below are the requirements for the Password Validator from the README for quick reference
+Your task is to create a `validatePassword` function which will return `true` when the password meets all requirements.
+
+- minimum of 8 characters
+
+Must contain at least one
+
+- lower case letter
+- upper case letter
+- numeric value
+- special character */
+
+module.exports = function validatePassword(password) {
```

Comment:
This is good. It looks like you may have done more work than was required.

Line 15 is the key part that making the first test pass.  Is the `for` loop which the `(password.length >= 8)` predicate is within required for `(password.length >= 8)` to work?

Regarding the looping logic and the name `initialChar`: This variable could be worth considering options for other names. It is important to keep in mind that when a variable is declared for an array index such was what is here, that variable is a numeric type. It does not contain a "char".  It contains a number.

Reply:
Hi, JR - thanks! The for loop was added after. I had forgot to commit the code with just the password.length condition. But now I'm trying to work on searching the string for the other rules. Proving to be quite the challenge. I was thinking of trying to get it to loop through every character within the password to search for an instance of each rule. Is this not a necessary step? Can I just search in one go without having to loop through?